### ISSUE TITLE
'no viable alternative at input' when designing binders

### ISSUE THREAD

**Issue by jbrydon2** [RosettaCommons/RFdiffusion#100]
Hello, 

I've got the inference script running successfully using the monomer example provided in the readme, and have also performed motif scaffolding (using the same pdb input as below). When I try to design a binder, I get the following:

(SE3nv) C:\Users\jbrydon2\RFdiffusion>python ./scripts/run_inference.py inference.input_pdb=DxL_Fc.pdb contigmap.contigs=[Z6-37/0 50-50] inference.output_prefix=bindertests inference.num_designs=10
no viable alternative at input '[Z6-37/0'
See https://hydra.cc/docs/1.2/advanced/override_grammar/basic for details

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.

It seems that using /0 to signify a chain break is causing a problem, but the syntax I've used is identical to the readme unless I am missing something. Thanks for your help - I am a complete beginner so please let me know if further information is needed!

**Comment by jbrydon2** [comment]
I have managed to overcome the error by replacing the space in the contigmaps config with a comma:

python ./scripts/run_inference.py inference.input_pdb=DxL_Fc.pdb contigmap.contigs=[50-50,z6-37/0] inference.output_prefix=bindertests inference.num_designs=10

Which allows the script to run though the timesteps, but then delivering this error:

...
[2023-07-28 12:42:08,807][rfdiffusion.inference.model_runners][INFO] - Timestep 2, input to next step: --------------------------------------------------
Error executing job with overrides: ['inference.input_pdb=DxL_Fc.pdb', 'contigmap.contigs=[50-50,z6-37/0]', 'inference.output_prefix=bindertests', 'inference.num_designs=10']
Traceback (most recent call last):
  File "C:\Users\jbrydon2\RFdiffusion\scripts\run_inference.py", line 122, in main
    os.makedirs(os.path.dirname(out_prefix), exist_ok=True)
  File "C:\Users\jbrydon2\AppData\Local\anaconda3\envs\SE3nv\lib\os.py", line 225, in makedirs
    mkdir(name, mode)
FileNotFoundError: [WinError 3] The system cannot find the path specified: ''

Does anyone know what exactly is the problem here? As I have said I have run the program for monomer and motif scaffolding, and had no issue writing the results to a premade folder. In this case I made the folder bindertests within the RF directory exactly as before, but it seems like there is some problem trying to write the results?

Thanks again.

**Comment by ddemonte0306** [comment]
In your first comment you use an uppercase "Z", and in the second a lowercase.  Does case matter here?

Assuming lowercase, why not place chain breaks after both (without comma):
contigmap.contigs=[50-50/0 z6-37/0]

**Comment by jbrydon2** [comment]
Thanks for your input - as far as I can tell case does not matter for the chain identifier - when I use a comma the script runs with either a lower or upper case Z. Adding a chain break for both chains didn't solve the problem when using a space, i.e.:

contigmap.contigs=[50-50/0,z6-37/0]

the above works, but the below does not

contigmap.contigs=[50-50/0 z6-37/0]

I am still having the 'FileNotFoundError: [WinError3]' described above, so anyone who has any insight would be much appreciated.

Thanks!

**Comment by jbrydon2** [comment]
Have now fixed this issue. My previous syntax had no quote marks around the contigmaps.contig hydra config, as each time I placed a single quote mark it returned the error:

LexerNoViableAltException: '
                                            ^

By removing the single quotes the script can be made to run, however this caused the arguments within the contigmap.contigs hydra to be incorrectly parsed when a whitespace is present between the two contig arguments. I replaced the single quote marks with double quote marks, e.g.:

(SE3nv) C:\Users\jbrydon2\RFdiffusion>python ./scripts/run_inference.py "contigmap.contigs=[z12-23/0 20-20]" inference.output_prefix=test_outputs/binder_tests inference.num_designs=1 inference.input_pdb=DxL_Fc.pdb

and this seemed to solve the issue, giving an output file with two separate chains.
